Add quiz to intro sorting

TruaShamu opened the pull request:
Not sure if I formatted this correctly - it looked alright in the editor, but if the preview doesn't work, I'll try again to fix the indentation

nwatx commented on content/2_Bronze/Intro_Sorting.mdx:
```diff
@@ -317,3 +317,72 @@ always suffice).
+	<Quiz.Answer>
+	  Bubble sort
+	  <Quiz.Explanation>
+		Incorrect.
```
an  explanation for these would be nice

nwatx commented on content/2_Bronze/Intro_Sorting.mdx:
```diff
@@ -317,3 +317,72 @@ always suffice).
+	  </Quiz.Explanation>
+	</Quiz.Answer>
+	<Quiz.Answer>
+	  Selection sort
```
add explanation

nwatx reviewed (changes requested):
it would also be nice to add a question like "Can … be sorted by (some criteria)?" and if it does not satisfy the transitive property then it can't be sorted, so that way people know what can or can't be sorted.

SansPapyrus683 commented:
![image](https://user-images.githubusercontent.com/55369003/135952443-16d76b01-a2c5-421b-b180-6f7cc968ceaa.png)
wait what does unstable even mean lmao

SansPapyrus683 commented:
and the last question is kinda weird
idk but smth about complexities would be better?

TruaShamu commented:
@nwatx
* I don't really understand -  so I know that if something is transitive, if a < b  and b < c, then a < c...
But are there arrays or data that isn't transitive? (Sorry, I'm dumb)
* I added explanations, hopefully the new version is somewhat better?
@SansPapyrus683 
* Idk if this is a clear explanation, but stable basically means - if there's 2 elements with same value, then after sorting, the order of the elements won't change. If you have any better suggestions for the problem, I could delete this problem instead?
* Also, I changed the 3rd problem to one about merge sort complexity, hopefully this is better?

TruaShamu commented:
Alright, thanks!
I'll add the definition to the problem for now - I'll probably add more content to the module later because I have a bunch of homework that I'm working on rn

SansPapyrus683 commented:
yeah alr i'm good

TruaShamu commented:
@nwatx 
Sorry for inconveniencing you, do you mind sharing the code with me again?
I know that if the custom comparator violates transitivity, there will be timsort exception 'Comparison method violates its general contract!'

I've been trying to implement a custom comparator by myself for reference, but I haven't been able to reproduce the exception.
It would be really helpful if you could resend me the code from the usaco guide ide, thanks

nwatx commented:
> @nwatx Sorry for inconveniencing you, do you mind sharing the code with me again? I know that if the custom comparator violates transitivity, there will be timsort exception 'Comparison method violates its general contract!'
> 
> I've been trying to implement a custom comparator by myself for reference, but I haven't been able to reproduce the exception. It would be really helpful if you could resend me the code from the usaco guide ide, thanks

oops, define two pairs and compare them by `(pair a, pair b) -> {return a.f - b.s < b.a - a.s;}`

TruaShamu commented:
Thank you.

nwatx commented:
also, almost done! :+1:

TruaShamu commented:
What is texify?

SansPapyrus683 commented:
yeah other than this it looks nice
